docs/source/cuda-reference/kernel.rst

```diff
@@ -62,16 +82,89 @@ Thread Indexing
+   the given ``shape`` and ``dtype``.
+
+   Returns an array with its content uninitialized.
+
+   .. note:: All threads in the same thread block sees the same array.
+
+.. function:: numba.cuda.local.array(shape, dtype)
+
+   Creates an array in the local memory space of the CUDA kernel with the
+   given ``shape`` and ``dtype``.
+
+   Returns an array with its content uninitialized.
+
+   .. note:: Each thread sees a unique array.
+
+.. function:: numba.cuda.const.array_like(ary)
+
+   Copies the ``ary`` into constant memory space on the CUDA kernel at compile
+   time.
+
+   Returns an array like the ``ary`` argument.
+
+   .. note:: All threads and blocks see the same array.
 
 Synchronization and Atomic Operations
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. autoclass:: numba.cuda.atomic
-   :members: add, max
-.. autofunction:: numba.cuda.syncthreads
+.. function:: numba.cuda.atomic.add(array, idx, value)
+
+    Perform ``array[idx] += value``. Support int32, int64, float32 and
+    float64 only. The ``idx`` argument can be an integer or a tuple of integer
+    indices for indexing into multiple dimensional arrays. The number of element
+    in ``idx`` must match the number of dimension of ``array``.
+
+    Returns the value of ``array[idx]`` before the storing the new value.
+    Behaves like an atomic load.
+
+.. function:: numba.cuda.atomic.max(array, idx, value)
+
+    Perform ``array[idx] = max(array[idx], value)``. Support int32, int64,
+    float32 and float64 only. The ``idx`` argument can be an integer or a
+    tuple of integer indices for indexing into multiple dimensional arrays.
+    The number of element in ``idx`` must match the number of dimension of
+    ``array``.
+
+    Returns the value of ``array[idx]`` before the storing the new value.
+    Behaves like an atomic load.
+
+
+.. function:: numba.cuda.syncthreads
+
+    Synchronize all threads in the same thread block.  This function implements
+    the same pattern as barriers in traditional multi-threaded programming: this
+    function waits until all threads in the block call it, at which point it
+    returns control to all its callers.
+
+    .. warning:: Must be called by every thread in the thread-block. Falling to
+                 do so may result in undefined behavior.
+
+Memory Fences
+~~~~~~~~~~~~~~
+
+The memory fences are used to guarantee the effect of memory operations
+are visible by other threads within the same thread-block, the same GPU device,
+and the same system (across GPUs on global memory). Memory loads and stores
+are guaranteed to not move across the memory fences by optimization passes.
+
+.. warning:: The memory fences are considered to be advanced API and most
+             usercases should use the thread barrier (e.g. ``syncthreads()``).
+
+
+
+.. function:: numba.cuda.threadfence
+
+   A memory fence at thread block level
+
+.. function:: numba.cuda.threadfence_block
+
+   A memory fence at thread block level.
```

Comment:
I think the description for `numba.cuda.threadfence` can read "A memory fence at the device level"